python/lsst/daf/butler/core/config.py

```diff
@@ -75,7 +77,11 @@ class Loader(yaml.CSafeLoader):
 
     def __init__(self, stream):
         super().__init__(stream)
-        self._root = ButlerURI(stream.name)
+        # if this is a string and not a stream we may well lack a name
+        try:
+            self._root = ButlerURI(stream.name)
+        except AttributeError:
+            self._root = None
```

Comment:
I don't think a second class is possible is it? This is a YAML plugin to handle `!include`.

Also, "stream" is what YAML calls it so I pretty much have to call it that.

Reply:
If you mean the fact that `Loader` has `yaml.CSafeLoader` as a base class, it certainly should not constrain your parameter names, and I don't think it would constrain whether `Loader` itself has subclasses -- especially if you're relying on factory methods everywhere.

Reply:
I think that this argument being passed directly to the YAML loader suggests that `stream` is the right name.